### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by jadibdev**
### Description
Refactored code observing YAGNI, KISS, and DRY principles as well as Single Responsibility.

### CODE REVIEW

**Review by FrederickMih — COMMENTED** [review]
Hi Team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Your README.md file is missing some sections like the **Build With** and **Getting Started**. Kindly follow the layout of this [Microverse README template](https://github.com/microverseinc/readme-template). 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app.rb [hunk 1]
```diff
@@ -48,30 +48,18 @@ def create_a_teacher
     @people.push(teacher)
   end
```
Comment (HAMMAS-SALEEM) [inline app.rb]: ## Your Functions
### 1.
DRY2 (copy)

### 2.
DRY3 (copy)

### 3.

DRY (copy)


- [ ] Please make your code DRY as as i suggested in the images above :+1:
- [ ] Kindly follow the SOLID principal means each function should have only one responsibility :+1:

File: app.rb [hunk 2]
```diff
@@ -100,6 +88,27 @@ def create_rental
     @rentals.push(rental)
   end
```
Comment (HAMMAS-SALEEM) [inline app.rb]: KISS1 (copy)

- [ ] Kindly follow the SOLID principal means each function should have only one responsibility as i suggested in the image above :+1:
- [ ] Kindly, make your code according to KISS principal which measn Keep it simple. But this function is not simple, you can make it simple by dividing the code into different functions :+1:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

### Hey @jadibdev, @Chrissiku :wave:

### Highlights :1st_place_medal:
 
:heavy_check_mark: Meaningful commits
:heavy_check_mark: Added .gitignore
:heavy_check_mark: Following GitFlow
:heavy_check_mark: Descriptive PR Title
:heavy_check_mark: Following Linters

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional Suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @HAMMAS-SALEEM in your question so I can receive the notification. Otherwise reach out to me on slack which is more fast. You can reach out to me on slack by searching my name in **full-time-students** channel :+1:

_**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_

------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by victorokoroji — APPROVED** [review]
Hi @jadibdev, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- Code is clean and dry✅ 
- The PR title and description are professional and descriptive ✅  
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Chrissiku** [comment]
Thank you @vickymarz, @FrederickMih  and @HAMMAS-SALEEM

**Pull request merged**